logdetective/server/database/models/merge_request_jobs.py

```diff
@@ -0,0 +1,473 @@
+    ) -> int:
+        """Create a new merge request job entry,
+        returns its PostgreSQL id
+
+        Args:
+          forge: forge name
+          project_id: forge project id
+          mr_iid: merge request forge iid
+          job_id: forge job id
+        """
+        with transaction(commit=True) as session:
+            mr = cls()
+            mr.forge = forge
+            mr.project_id = project_id
+            mr.mr_iid = mr_iid
+            mr.job_id = job_id
+            session.add(mr)
+            session.flush()
+            return mr.id
+
+    @classmethod
+    def get_by_id(
+        cls,
+        id_: int,
+    ) -> Optional["GitlabMergeRequestJobs"]:
+        """Search for a given PostgreSQL id"""
+        with transaction(commit=False) as session:
+            mr = session.query(cls).filter_by(id=id_).first()
+            return mr
+
+    @classmethod
+    def get_by_details(
+        cls,
+        forge: Forge,
+        project_id: int,
+        mr_iid: int,
+        job_id: int,
+    ) -> Optional["GitlabMergeRequestJobs"]:
+        """Search for a detailed merge request.
+
+        Args:
+          forge: forge name
+          project_id: forge project id
+          mr_iid: merge request forge iid
+          job_id: forge job id
+        """
+        with transaction(commit=False) as session:
+            mr = (
+                session.query(cls)
+                .filter_by(
+                    forge=forge, project_id=project_id, mr_iid=mr_iid, job_id=job_id
+                )
+                .first()
+            )
+            return mr
+
+    @classmethod
+    def get_or_create(
+        cls,
+        forge: Forge,
+        project_id: int,
+        mr_iid: int,
+        job_id: int,
+    ) -> Optional["GitlabMergeRequestJobs"]:
+        """Search for a detailed merge request
+        or create a new one if not found.
+
+        Args:
+          forge: forge name
+          project_id: forge project id
+          mr_iid: merge request forge iid
+          job_id: forge job id
+        """
+        mr = GitlabMergeRequestJobs.get_by_details(forge, project_id, mr_iid, job_id)
+        if mr is None:
+            id_ = GitlabMergeRequestJobs.create(forge, project_id, mr_iid, job_id)
+            mr = GitlabMergeRequestJobs.get_by_id(id_)
+        return mr
+
+
+class Comments(Base):
+    """Store details for comments
+    created by logdetective for a merge request job."""
+
+    __tablename__ = "comments"
+
+    id = Column(Integer, primary_key=True)
+    merge_request_job_id = Column(
+        Integer,
+        ForeignKey("gitlab_merge_request_jobs.id"),
+        nullable=False,
+        unique=True,  # 1 comment for 1 job
+        index=True,
+        comment="The associated merge request job (db) id",
+    )
+    comment_id = Column(
+        String(50),  # e.g. 'd5a3ff139356ce33e37e73add446f16869741b50'
+        nullable=False,
+        index=True,
+        comment="The comment gitlab id",
+    )
```

Comment:
The `comment_id` is unique within a forge instance. Can you define that constraint, please?